quantus/metrics/base_batched.py

```diff
@@ -0,0 +1,404 @@
+# Quantus is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+# You should have received a copy of the GNU Lesser General Public License along with Quantus. If not, see <https://www.gnu.org/licenses/>.
+# Quantus project URL: <https://github.com/understandable-machine-intelligence-lab/Quantus>.
+
+import inspect
+import math
+import re
+from abc import abstractmethod
+from typing import Any, Callable, Dict, Optional, Sequence, Union
+
+import numpy as np
+from tqdm.auto import tqdm
+
+from .base import Metric
+from ..helpers import asserts
+from ..helpers import warn_func
+from ..helpers.model_interface import ModelInterface
+
+
+class BatchedMetric(Metric):
+    """
+    Implementation base BatchedMetric class.
+    """
+
+    @asserts.attributes_check
+    def __init__(
+        self,
+        abs: bool,
+        normalise: bool,
+        normalise_func: Optional[Callable],
+        normalise_func_kwargs: Optional[Dict[str, Any]],
+        return_aggregate: bool,
+        aggregate_func: Optional[Callable],
+        default_plot_func: Optional[Callable],
+        disable_warnings: bool,
+        display_progressbar: bool,
+        **kwargs,
+    ):
+        """
+        Initialise the Metric base class.
+
+        Each of the defined metrics in Quantus, inherits from Metric base class.
+
+        A child metric can benefit from the following class methods:
+        - __call__(): Will call general_preprocess(), apply evaluate_instance() on each
+                      instance and finally call custom_preprocess().
+                      To use this method the child Metric needs to implement
+                      evaluate_instance().
+        - general_preprocess(): Prepares all necessary data structures for evaluation.
+                                Will call custom_preprocess() at the end.
+
+        Parameters
+        ----------
+        abs (boolean): Indicates whether absolute operation is applied on the attribution.
+        normalise (boolean): Indicates whether normalise operation is applied on the attribution.
+        normalise_func (callable): Attribution normalisation function applied in case normalise=True.
+        normalise_func_kwargs (dict): Keyword arguments to be passed to normalise_func on call.
+        default_plot_func (callable): Callable that plots the metrics result.
+        disable_warnings (boolean): Indicates whether the warnings are printed.
+        display_progressbar (boolean): Indicates whether a tqdm-progress-bar is printed.
+
+        """
+        # We need to do this separately here, as super().__init__() overwrites these
```

Comment:
What does "do this" mean in "We need to do this separately here"

Reply:
i have removed this custom kwargs adding as we now dont change the evaulate method signature.